### PR TITLE
Initial work on tidyselect (#209)

### PR DESCRIPTION
**Pull request by wlandau**
Implement `tidyselect` in `loadd()` and `clean()` so the following is now possible
```r
load_basic_example(verbose = FALSE)
make(my_plan, verbose = FALSE)
cached(no_imported_objects = TRUE)

e <- new.env()
ls(e)

# character(0)

loadd(ends_with("small"), envir = e)
ls(e)

## [1] "coef_regression1_small" "coef_regression2_small" "regression1_small"
## [4] "regression2_small"      "small"                  "summ_regression1_small"
## [7] "summ_regression2_small"

cached(no_imported_object = TRUE)

##  [1] "'report.md'"            "'report.Rmd'"           "coef_regression1_large"
##  [4] "coef_regression1_small" "coef_regression2_large" "coef_regression2_small"
##  [7] "large"                  "regression1_large"      "regression1_small"
## [10] "regression2_large"      "regression2_small"      "small"
## [13] "summ_regression1_large" "summ_regression1_small" "summ_regression2_large"
## [16] "summ_regression2_small"

cached(no_imported_object = TRUE)

##  [1] "'report.md'"            "'report.Rmd'"           "large"
##  [4] "regression1_large"      "regression1_small"      "regression2_large"
##  [7] "regression2_small"      "small"                  "summ_regression1_large"
## [10] "summ_regression1_small" "summ_regression2_large" "summ_regression2_small"
```

@krlmlr Am I doing this right?

### CODE REVIEW

File: tests/testthat/test-basic.R [hunk 1]
```diff
@@ -92,8 +92,25 @@ test_with_dir("basic example works", {
...
+  clean(starts_with("coef"))
+  expect_equal(sort(cached()), setdiff(x, coefs))
+
   # knitr file deps
   # Included here instead of test-knitr.R because report.md already exists.
+  # Saves time that way.
+  # But we can remove it all when we deprecate the single-quoted stuff
+  # and users know to rely on knitr_input().
   x <- drake_plan(
     a = knitr::knit(knitr_input(report.Rmd)), # nolint
```
Comment (krlmlr) [inline tests/testthat/test-basic.R]: Are we allowed to omit quotes now?

File: R/tidyselect_reexport.R [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+tidyselect::ends_with
+#' @importFrom tidyselect everything
+#' @export
+tidyselect::everything
+#' @importFrom tidyselect matches
+#' @export
+tidyselect::matches
+#' @importFrom tidyselect num_range
+#' @export
+tidyselect::num_range
```
Comment (krlmlr) [inline R/tidyselect_reexport.R]: Do we really need `num_range()`? Are our targets ordered?

File: R/tidyselect_reexport.R [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# From `dplyr`:
+# Flag to disable hotpatching from old tidyselect versions
+peek_vars <- tidyselect::peek_vars
```
Comment (krlmlr) [inline R/tidyselect_reexport.R]: I don't think we need this here.

File: tests/testthat/test-basic.R (hunk 1 above)
Comment (wlandau) [reply tests/testthat/test-basic.R]: Nope, my mistake. See 97ad2b59ffdab3c5d09664c89a82d845bc24dbdc.

File: R/tidyselect_reexport.R (hunk 2 above)
Comment (wlandau) [reply R/tidyselect_reexport.R]: I was just copying the re-exports of `dplyr`, and I am new to this. However, I do think there is a use for `num_range()`. Suppose people want to use numbers to index their targets?

```r
x <- drake_plan(a = rep) %>% 
  evaluate_plan(wildcard = "rep", values = 1:3)
make(x)
loadd(num_range(prefix = "a_", range = 1:3))
```

File: R/tidyselect_reexport.R (hunk 3 above)
Comment (wlandau) [reply R/tidyselect_reexport.R]: What does `peek_vars()` actually do? Does it help with tidyselect, or did dplyr need it for another reason?

**Comment by wlandau** [comment]
I cannot reproduce the current test error locally, and I cannot find any reference to `lang_unnamespace()` in a Google search. Pinging [StackOverflow](https://stackoverflow.com/questions/48893121/error-could-not-find-function-lang-unnamespace). Without a minimal reproducible example, it is not likely to end well.

File: R/tidyselect_reexport.R (hunk 3 above)
Comment (krlmlr) [reply R/tidyselect_reexport.R]: No idea. Since we're not using it, we don't need to import and fix it.

**Comment by krlmlr** [comment]
Can you try clearing the Travis CI cache?

**Comment by krlmlr** [comment]
You can also enable debugging for Travis CI builds and use `travis::travis_debug_job()` (from ropenscilabs/travis).

**Comment by wibeasley** [comment]
>Without a minimal reproducible example, the thread is not likely to end well.

fwiw, I think the link to the travis error is probably even more helpful.  Hope this eventually helps a little: 
 https://stackoverflow.com/questions/48893121/error-could-not-find-function-lang-unnamespace/48893591#48893591

File: R/tidyselect_reexport.R (hunk 3 above)
Comment (wlandau) [reply R/tidyselect_reexport.R]: Removed in 97ad2b59ffdab3c5d09664c89a82d845bc24dbdc. Added `last_col()` just for the sake of making all the `tidyselect::vars_select_helpers` to the users.

**Comment by wlandau** [comment]
The last commit had no build problems. From @wibeasley on StackOverflow, this may have been caused by an update to `rlang`. The unfinished CRAN builds may have somehow confused Travis. Thanks again for the advice.

**Comment by wlandau** [comment]
The Travis error went away, but now I am [getting it on AppVeyor](https://ci.appveyor.com/project/ropensci/drake/build/1.0.564)

**Pull request merged**